pkg/measurements/factory.go

```diff
@@ -73,16 +73,28 @@ func isIndexerOk(configSpec config.Spec, measurement types.Measurement) bool {
 }
 
 // NewMeasurementsFactory initializes the measurement facture
-func NewMeasurementsFactory(configSpec config.Spec, metadata map[string]interface{}) *MeasurementsFactory {
+func NewMeasurementsFactory(configSpec config.Spec, metadata map[string]interface{}, additionalMeasurementFactoryMap map[string]NewMeasurementFactory) *MeasurementsFactory {
+	measurementFactoryMap := make(map[string]NewMeasurementFactory)
+	//copy all entries from baseMeasurementFactoryMap
+	for k, v := range baseMeasurementFactoryMap {
+		measurementFactoryMap[k] = v
+	}
+	// Add from additionalMeasurementFactoryMap without overwriting
+	for k, v := range additionalMeasurementFactoryMap {
+		if _, exists := measurementFactoryMap[k]; !exists {
+			measurementFactoryMap[k] = v
+		}
+	}
+
```

Comment:
I think we don't need a new variable `baseMeasurementFactoryMap`, we can add the additional measurements directly to the previously defined measurementFactoryMap


```suggestion
	// Add from additionalMeasurementFactoryMap without overwriting
	for k, v := range additionalMeasurementFactoryMap {
		if _, exists := measurementFactoryMap[k]; !exists {
			measurementFactoryMap[k] = v
		}
	}

```